rotkehlchen/chain/ethereum/tokens.py

```diff
@@ -61,64 +64,92 @@
-        balances: Dict[EthereumToken, FVal] = defaultdict(FVal)
-        if self.ethereum.connected_to_any_web3():
-            # For all the chunks we use the same order of the nodes
-            call_order = self.ethereum.default_call_order()
-            for chunk in other_chunks:
-                self._get_tokens_balance_and_price(
-                    address=address,
-                    tokens=chunk,
-                    balances=balances,
-                    token_usd_price=token_usd_price,
-                    call_order=call_order,
-                )
-        else:
-            for chunk in etherscan_chunks:
-                self._get_tokens_balance_and_price(
-                    address=address,
-                    tokens=chunk,
-                    balances=balances,
-                    token_usd_price=token_usd_price,
-                    call_order=(ETHERSCAN_NODE,),
-                )
+        """Queries the balances of multiple tokens for an account
 
-        # now that detection happened we also have to save it in the DB for the address
-        self.db.save_tokens_for_address(write_cursor, address, list(balances.keys()))
+        May raise:
+        - RemoteError if an external service such as Etherscan is queried and
+          there is a problem with its query.
+        - BadFunctionCallOutput if a local node is used and the contract for the
+          token has no code. That means the chain is not synced
+        """
+        log.debug(
+            'Querying ethereum chain for multi token account balances',
+            eth_address=account,
+            tokens_num=len(tokens),
+        )
+        result = ETH_SCAN.call(
+            ethereum=self.ethereum,
+            method_name='tokensBalance',
+            arguments=[account, [x.ethereum_address for x in tokens]],
+            call_order=call_order,
+        )
+        balances: Dict[EthereumToken, FVal] = defaultdict(FVal)
+        for token_balance, token in zip(result, tokens):
+            if token_balance == 0:
+                continue
 
+            normalized_balance = token_normalized_value(token_balance, token)
+            log.debug(
+                f'Found {token.symbol}({token.ethereum_address}) token balance for '
+                f'{account} and balance {normalized_balance}',
+            )
+            balances[token] += normalized_balance
         return balances
 
-    def query_tokens_for_addresses(
-            self,
-            addresses: List[ChecksumEthAddress],
-            force_detection: bool,
-    ) -> TokensReturn:
-        """Queries/detects token balances for a list of addresses
-
-        If an address's tokens were recently autodetected they are not detected again but the
-        balances are simply queried. Unless force_detection is True.
+    def _query_chunks(
+        self,
+        account: ChecksumEthAddress,
+        chunks: Iterable[List[EthereumToken]],
+        querying_etherscan: bool,
```

Comment:
Please configure your editor to follow the indentation style. + 1 indentation on these arguments.

```suggestion
        self,
        account: ChecksumEthAddress,
        chunks: Iterable[List[EthereumToken]],
        querying_etherscan: bool,
```

Reply:
Ok